### PR TITLE
[WIP] Enable AI.SCRIPTRUN on AI.DAGRUN*

### LINKED ISSUE #382: AI.SCRIPTRUN results are being replicated twice
The results are being replicated on `RAI_ModelRunScriptRunReply` via `RedisAI_ReplicateTensorSet` and via `RedisModule_ReplicateVerbatim` within `RedisAI_ScriptRun_RedisCommand`. 
I believe the call that should be removed is the `RedisModule_ReplicateVerbatim` given that `RedisAI_ReplicateTensorSet` will ensure determinism on slves in cases of on determisitic outputs ( like we have on modelrun )

### PR DESCRIPTION
**Pull request by filipecosta90**
This is a WIP PR for including AI.SCRIPTRUN within the DAG execution ( feel free to jump in @lantiga :) )
- [x] decoupled scriptrun command parsing from scriptrun command
- [x] splitted positive/negative tests on pytorch scriptrun. 
- [x] refactor AI.DAGRUN_RO and AI.DAGRUN to use the same code base (with read/write modes)
- [x] include SCRIPTRUN on DAGRUN and DAGRUN_RO
- [x] extensive testing for AI.SCRIPTRUN within AI.DAGRUN
- [x] update documentation to reflect scriptrun capabilities in DAGRUN

### CODE REVIEW

File: src/dag.c [hunk 1]
```diff
@@ -325,3 +373,163 @@ int RAI_parseDAGPersistArgs(RedisModuleCtx *ctx, RedisModuleString **argv,
   }
   return argpos;
 }
+
+int RedisAI_DagRun_syntax_parser(RedisModuleCtx *ctx, RedisModuleString **argv,
```
Comment (lantiga) [inline src/dag.c:377]: We should be more consistent with casing / underscores for function names throughout the module

Comment (filipecosta90) [reply src/dag.c:377]: agree :) changing to `RedisAI_DagRunSyntaxParser`.

File: docs/commands.md [hunk 2]
```diff
@@ -561,6 +562,30 @@ redis> AI.DAGRUN PERSIST 1 predictions |>
    3) "\x00\x00\x80?\x00\x00\x00@\x00\x00@@\x00\x00\x80@"
 ```
 
+A common pattern is enqueuing multiple SCRIPTRUN and MODELRUN commands within a DAG. The following example uses ResNet-50,to classify images into 1000 object categories. Given that our input tensor contains each color represented as a 8-bit integer and that neural networks usually work with floating-point tensors as their input we need to cast a tensor to floating-point and normalize the values of the pixels - for that we will use `pre_process_3ch` function. 
+
+To optimize the classification process we can use a post process script to return only the category position with the maximum classification - for that we will use `post_process` script. Using the DAG capabilities we've removed the necessity of storing the intermediate tensors in the keyspace. You can even run the entire process without storing the output tensor, as follow:
```
[the remark this answers is not in the document]
Comment (filipecosta90) [reply docs/commands.md]: solved in the just pushed commit

File: src/dag.c [hunk 3]
```diff
@@ -325,3 +373,163 @@ int RAI_parseDAGPersistArgs(RedisModuleCtx *ctx, RedisModuleString **argv,
...
+        if (deviceStr == NULL) {
+          deviceStr = sto->devicestr;
+        } else {
+          // If the device strings are not equivalent, reply with error ( for
+          // now )
+          if (strcasecmp(sto->devicestr, deviceStr) != 0) {
+            RAI_FreeRunInfo(ctx, rinfo);
+            return RedisModule_ReplyWithError(
+                ctx, "ERR multi-device DAGs not supported yet");
+            ;
```
[the remark this answers is not in the document]
Comment (filipecosta90) [reply src/dag.c]: solved in the just pushed commit

**Comment by filipecosta90** [comment]
@lantiga I'll split the fix for #382 (AI.SCRIPTRUN results are being replicated twice) from this added feature. It  makes sense to not mix them.
